README.md

```diff
@@ -15,70 +15,77 @@
 [s6]: https://tokei.rs/b1/github/amethyst/laminar?category=code
 [s7]: https://codecov.io/gh/amethyst/laminar/branch/master/graphs/badge.svg
 
-A UDP-based protocol that provides partial reliability. Coming soon!
+This library implements some TCP-like features on top of an UDP-socket. 
+I will provide a lightweight message-based interface with certain guarantees like reliability, fragmentation congestion monitoring.
 
-## Note
+Laminar was designed to be used in the [Amethyst][amethyst] game engine and is mostly based on articles from [gaffer on games](https://gafferongames.com/).
 
-This library is not yet stable. It is experimental and things may change frequently.
+[amethyst]: https://github.com/amethyst/amethyst
 
 ## Table of contents:
-- [Useful links](https://github.com/amethyst/laminar#useful-links)
-- [Features](https://github.com/amethyst/laminar#features)
-- [Examples](https://github.com/amethyst/laminar#examples)
-    - [Udp](https://github.com/amethyst/laminar#udp)
-- [Notice](https://github.com/amethyst/laminar#notice)
-- [Contributing](https://github.com/amethyst/laminar#contributing)
-- [Authors](https://github.com/amethyst/laminar/#authors)
+- [Useful links](#useful-links)
+- [Features](#features)
+- [Getting Started](#getting-stated)
+- [Examples](#examples)
+- [Notice](#notice)
+- [Contributing](#contribution)
+- [Authors](#authors)
 - [License](#license)
 
+## Features
+These are the features this crate provides:
+
+- UDP-based protocol
+- Automatic Fragmentation
+- RTT estimation
+- Connection tracking.
+- Unreliable and Reliable sending of packets.
+- Protocol version monitoring.
+- A link conditioner to simulate packet loss and latency
+- Good error handling with **zero** panics.
+- Well tested by integration tests and unit tests
+- Benchmarks
+
+## Getting Stated
 Add the laminar package to your `Cargo.toml` file.
 
 ```toml
 [dependencies]
-laminar = "0.0.0"
-```
-And import the laminar modules you want to use.
-
-```rust
-extern crate laminar;
-
-// this module contains all socket related logic.
-use laminar::net::{UdpSocket, SocketAddr, NetworkConfig, Connection, Quality};
-// this module contains packet related logic.
-use laminar::packet::{Packet};
+laminar = "0.1.0"
 ```
 
-## Useful Links
+### Useful Links
 
 - [Documentation](https://docs.rs/laminar/).
 - [Cargo Page](https://crates.io/crates/laminar)
 - [Examples](https://github.com/amethyst/laminar/tree/master/examples)
-
-## Features
-These are the features from this crate:
-
-- Semi-reliable UDP
-- Fragmentation
-- RTT estimation
-- Virtual connection management.
+- [Contributing Guides](https://github.com/amethyst/laminar/blob/master/docs/StyleGuides)
```

Comment:
We should move this to `CONTRIBUTING.md` and base that one off of the main repo.